pkg/rancher-desktop/config/commandLineOptions.ts

```diff
@@ -0,0 +1,151 @@
+import { join } from 'path';
+
+import _ from 'lodash';
+
+import {
+  getObjectRepresentation, getUpdatableNode, LockedSettingsType, save, Settings, turnFirstRunOff,
+} from '@pkg/config/settings';
+import { TransientSettings } from '@pkg/config/transientSettings';
+import SettingsValidator from '@pkg/main/commandServer/settingsValidator';
+import Logging from '@pkg/utils/logging';
+import paths from '@pkg/utils/paths';
+import { RecursiveKeys, RecursivePartial } from '@pkg/utils/typeUtils';
+
+const console = Logging.settings;
+
+export class LockedFieldError extends Error {}
+
+/**
+ * Takes an array of strings, presumably from a command-line used to launch the app.
+ * Key operations:
+ * * All options start with '--'.
+ * * Ignore leading unrecognized options.
+ * * Complain about any unrecognized options after a recognized option has been processed.
+ * * This calls the same settings-validator as used by `rdctl set` and the API to catch
+ *   any attempts to update a locked field.
+ *
+ *  * All errors are fatal as this function is like an API for launching the application.
+ * @param cfg
+ * @param lockedFields
+ * @param commandLineArgs
+ * @return updated cfg
+ */
+export function updateFromCommandLine(cfg: Settings, lockedFields: LockedSettingsType, commandLineArgs: string[]): Settings {
+  const lim = commandLineArgs.length;
+  let processingExternalArguments = true;
+  const updatedCfg = _.merge({}, cfg);
+  let newSettings: RecursivePartial<Settings> = {};
+
+  // As long as processingExternalArguments is true, ignore anything we don't recognize.
+  // Once we see something that's "ours", set processingExternalArguments to false.
+  // Note that `i` is also incremented in the body of the loop to skip over parameter values.
+  for (let i = 0; i < lim; i++) {
+    const arg = commandLineArgs[i];
+
+    if (!arg.startsWith('--')) {
+      if (processingExternalArguments) {
+        continue;
+      }
+      throw new Error(`Unexpected argument '${ arg }' in command-line [${ commandLineArgs.join(' ') }]`);
+    }
+    const equalPosition = arg.indexOf('=');
+    const [fqFieldName, value] = equalPosition === -1 ? [arg.substring(2), ''] : [arg.substring(2, equalPosition), arg.substring(equalPosition + 1)];
```

Comment:
So this is a complicated function that's been in place for a while, and again I think it would be better to create an issue to review it rather than add redoing it into this PR